src/main/java/com/garbagemule/MobArena/ArenaListener.java

```diff
@@ -1198,8 +1198,8 @@ else if (region.contains(to)) {
             if (p.hasPermission("mobarena.admin.teleport")) {
                 return TeleportResponse.ALLOW;
             }
-            
-            if (region.isWarp(from) || region.isWarp(to) || to.equals(arena.getPlayerEntry(p))) {
+
+            if ((arena.getAllPlayers().contains(event.getPlayer()) && (region.isWarp(from) || region.isWarp(to))) || to.equals(arena.getPlayerEntry(p))) {
```

Comment:
I have tested this practically before, it does not give any advantage. Garbagemule and I looked through the code a few minutes ago and found out why.

This "contains" check can only ever be true at this point in the code. This method gets the same lists as the individual checks in line 1184. Had one of them contained the player, the teleport would have been allowed already.